### PR TITLE
feat: driver-activity-report

### PR DESCRIPTION
**Pull request by Harmanfleet**
feat: driver-activity-report

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/pages/reports/fleet/drivers/driver-data/driver-data.component.ts [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+      this.driverID = `${firstName}-${middleName}-${lastName}`;
+    } else {
+      this.driverName = `${firstName} ${lastName}`;
+      this.driverID = `${firstName}-${lastName}`;
+    }
+
+    this.suggestedDrivers = [];
+  }
+
+  fetchDriversCount() {
```
Comment (kunalfleethawks) [inline src/app/pages/reports/fleet/drivers/driver-data/driver-data.component.ts]: Merge it into single api

File: src/app/pages/reports/fleet/drivers/driver-data/driver-data.component.ts [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+    this.apiService
+      .getData(
```
Comment (kunalfleethawks) [inline src/app/pages/reports/fleet/drivers/driver-data/driver-data.component.ts]: Please implement infinite paging

File: src/app/pages/reports/fleet/drivers/driver-data/driver-data.component.ts [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+    });
+  }
+  fetchAllCitiesIDs() {
+    this.apiService.getData("cities/get/list").subscribe((result: any) => {
+      this.citiesObject = result;
+    });
+  }
+  fetchAllVehiclesIDs() {
+    this.apiService.getData("vehicles/get/list").subscribe((result: any) => {
+      this.vehiclesObject = result;
```
Comment (kunalfleethawks) [inline src/app/pages/reports/fleet/drivers/driver-data/driver-data.component.ts:192]: Merge into single api

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]
Please address review comments

**Review by FH-sandeepkaur1 — APPROVED** [review]

**Review by kunalfleethawks — APPROVED** [review]

**Pull request merged**